client/src/components/settings/about.tsx

```diff
@@ -259,7 +258,7 @@ class AboutSettings extends Component<AboutProps, AboutState> {
                 />
               </FormGroup>
             </div>
-            <BlockSaveButton disabled={this.isFormPristine()}>
+            <BlockSaveButton aria-disabled={this.isFormPristine()}>
```

Comment:
```suggestion
            <BlockSaveButton
              aria-disabled={this.isFormPristine()}
              {...(this.isFormPristine() && { 'tabindex': -1 })}
            >
```
@Sboonny, sorry about this late addition. I would like to suggest that we include `tabindex="-1"` on the Save button when it is disabled, which will prevent keyboard users from tabbing to it. I think it makes sense to keep them out of the natural tab order to avoid confusion. I have tested this with both NVDA and JAWS screen readers and this will still allow us to keep the focus on the Save button when the user clicks it to save their preferences and it goes back to being disabled.

Reply:
My concern about this, we do have multiple form, with different save buttons, maybe someone will fill the form and expect it to update two different data 💭, but be surprised that only one form is submitted, because they didn't interact with correctly with our forms. aka, different API interaction.

to make my point more accessible o(*￣▽￣*)ブ and easier to understand.

Let's focus on this section of the toggle buttons.

![Show toggle buttons sections in the setting](https://user-images.githubusercontent.com/88248797/218257267-34339f3a-859c-4f5e-b7f1-058870e8f752.png)

We can change `Enable Keyboard Shortcuts`, and get a visual indication, the `Alert`, that the API is called. 

But when we move to the `Privacy Setting` section and don't get the API call, we know that this section isn't like the one before it

My worry is that camper navigates to the bottom of privacy page, finish with their preferences

![image](https://user-images.githubusercontent.com/88248797/218257604-d2d0720d-f3dc-4521-8aea-e29fbc2e0c12.png)

And doesn't save, because they are expecting the API to auto save their preferences.

So my questions are:

- Do screen-reader get Alert of a successful API call, when API is triggered?
- And do you think `sr-only` text is good enough to reduce the confusion for campers, so they don't think that a `Save` buttons work for Multiple forms.?

---

outside the scope of this PR, but I think you deserve to know, my full plan, to make the best experience possible

I was thinking of disabling the button, when the input isn't valid too, then change the button text to something like `field {field.name} isn't valid, please edit it to save` 🤔.

But this is for the future, and it should be simple `{isValid ? change field: save}`, and adding `tabindex` shouldn't change it, because it will apply only when `disabled`.

Reply:
Just to clarify, adding `tabindex="-1"` to the button will still allow it to appear on the page. It just won't allow keyboard users to tab to it (to put keyboard focus on it). Which is the way it works now on production. So we aren't really changing any behavior by doing this.

Reply:
> My concern about this, we do have multiple form, with different save buttons, maybe someone will fill the form and expect it to update two different data thought_balloon, but be surprised that only one form is submitted, because they didn't interact with correctly with our forms.

I agree, this could potentially happen. But I don't think that's an issue to solve in this PR.

Reply:
True, I have misspoken. 

To clarify, I wanted to say that:

- I want the buttons to be shown, because there are areas that act the same but aren't the same, aka the toggle buttons.
- I don't know if screen-readers do receive an alert, like the visual users do, so maybe it's best to show the buttons, if they don't get notified when the API is called.

Reply:
> I don't know if screen-readers do receive an alert, like the visual users do

Yes, the alert message that is displayed at the top of the page is read automatically by screen readers. So I don't think there is any concern to be had there.

Reply:
Although I do think it might help if we made those save alerts a little more specific. For example, if you save a change in the Personal Information, then instead of the alert just saying "We have updated your preferences" it might be nice if it said "We have updated your Personal Information changes".

Reply:
Sure, we can do this 👍, I had a plan to update portfolio item alert message in another PR, I will include the personal alert in the PR there, as well.